### ISSUE TITLE
False positive for undefined variable

### ISSUE THREAD

**Issue by memeplex** [microsoft/pylance-release#977]
## Environment data

-   Language Server version:  v2021.2.3
-   OS and version: MacOS 11.2
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.6

## Expected behaviour

Given a module `test_module` in the python path defining an object `test_object` the following code correctly defines `to` as a global variable of the current module accessible from the scope of function `f` and referencing `test_object`.

```python
def f():
    global to
    from test_module import test_object as to

    to.x = 20
```

## Actual behaviour

Pylance reports undefined variable for `to`. Removing the `global` declaration "fixes" it, but I want `to` to:

1. Be global
2. Be imported upon `f` invocation, not before.

Besides it's valid code, despite the somewhat convoluted use case.

**Comment by memeplex** [comment]
Notice that the import renaming is not necessary, it also happens for:

```
def f():
    global test_object
    from test_module import test_object

    test_object.x = 20
```

What is critical is the global declaration.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.2.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202124-24-february-2021

**Comment by memeplex** [comment]
Thanks!
